Detectors/ITSMFT/common/reconstruction/src/Clusterer.cxx

```diff
@@ -165,13 +181,19 @@ void Clusterer::finishChip(TClonesArray &clusters)
     Point3D<float> xyzLoc( Segmentation::getFirstRowCoordinate() + x*Segmentation::PitchRow/npix, 0.f,
 			   Segmentation::getFirstColCoordinate() + z*Segmentation::PitchCol/npix );
     auto xyzTra = mGeometry->getMatrixT2L(mChipData.chipID)^(xyzLoc); // inverse transform from Local to Tracking frame
-    Cluster *c = static_cast<Cluster *>(clusters.ConstructedAt(noc++));
-    c->setROFrame(mChipData.roFrame);
-    c->setSensorID(mChipData.chipID);
-    c->setPos(xyzTra);
-    c->setErrors(SigmaX2, SigmaY2, 0.f);
-    c->setNxNzN(xmax-xmin+1,zmax-zmin+1,npix);
-    for (int i=nlab;i--;) c->setLabel(labels[i],i);
+
+    clusters.emplace_back();
+    Cluster &c = clusters[noc];
+    c.SetUniqueID(noc); // Let the cluster remember its position within the cluster array
```

Comment:
@iouribelikov: I assume once all detectors get rid of TClonesArrays we will eliminate also inheritance of BaseCluster from TObject. If we really want to store the ID of the cluster in itself, we better add a dedicated data member to it. But do we really need this? The only use case is when you iterate over the cluster pointers cashed in some containers (like in trackers) rather than over the clusters vector directly. Should not we simply fill (e.g. in tracker) a vector mapping cashed ID on position of the cluster in original array?

Reply:
Absolutely ! The next step is to remove TObject from the BaseCluster.
But, I would need to think a bit more what to do with the cluster-position ID...
We could add a dedicated data member for it, for the moment...